openpdf/src/main/java/com/lowagie/text/pdf/PdfDocument.java

```diff
@@ -651,22 +651,36 @@ public boolean add(Element element) throws DocumentException {
                     leading = listItem.getTotalLeading();
                     carriageReturn();
 
-                    // we prepare the current line to be able to show us the listsymbol
-                    line.setListItem(listItem);
-                    // we process the item
-                    element.process(this);
+                    // if a listitem has to be kept together, we wrap it in a table object
+                    if (listItem.getKeepTogether()) {
+                        carriageReturn();
+                        PdfPTable table = createInOneCell(listItem);
+                        indentation.indentLeft += listItem.getIndentationLeft();
+                        indentation.indentRight += listItem.getIndentationRight();
 
-                    addSpacing(listItem.getSpacingAfter(), listItem.getTotalLeading(), listItem.getFont());
+                        this.add(table);
```

Comment:
By calling `add` here, the new type is `TABLE` and in consequence we loose the list-symbols in the resulting PDF. We need to find a better way to process the table which actually preserves the functionality of the list item. You probably did not realise this since your test does not test the actual functionality (see other comment on the test).